docs/release_process.md

```diff
@@ -0,0 +1,319 @@
+# Release Process
+## Environments
+
+* [Production][prod] - Run by SRE team in GCP
+* [Stage][stage] - Run by SRE team in GCP
+* [Dev][dev] - Run by ENGR team in Heroku
+* Locals: Run by ENGRs on their own devices. (See [README][readme] and other [`docs/`][docs].)
+
+## Code branches
+Standard Monitor development follows a branching strategy similar to
+[GitHub Flow][github-flow], where all branches stem directly from `main` and
+are merged back to `main`:
+
+1. Create a branch from `main`
+2. Make changes
+3. Create a pull request to `main`
+4. Address review comments
+5. Merge pull request
+
+```mermaid
+%%{init: { 'theme': 'base', 'gitGraph': {'rotateCommitLabel': true} } }%%
+    gitGraph
+       commit
+       branch change-1
+       commit
+       commit
+       checkout main
+       merge change-1
+```
+
+This means many features could be in development at the same time, and all can
+merge back to `main` when they are ready.
+
+```mermaid
+%%{init: { 'theme': 'base', 'gitGraph': {'rotateCommitLabel': true} } }%%
+    gitGraph
+       commit
+       branch change-1
+       commit
+       commit
+       checkout main
+       branch change-2
+       commit
+       checkout main
+       merge change-1
+       branch change-3
+       commit
+       commit
+       checkout main
+       merge change-2
+       checkout change-3
+       commit
+       commit
+       checkout main
+```
+
+## Release Timeline
+The standard release interval for Monitor is 1 week, meaning every week there
+will be a new version of the Monitor web app on the [Production][prod]
+environment. To do this, we first release code to [Dev][dev] and
+[Stage][stage].
+
+## Release to Dev
+Every commit to `main` is automatically deployed to the [Dev][dev] server, as
+long as it can be done with a fast-forward push. Since the
+[Great GitHub Heroku Incident of 2022][github-heroku-incident], this is
+done from CircleCI using a [service account][service-account].
+
+To push a different branch, you need to add the Heroku app as a remote:
+
+* `heroku login`
+* `heroku git:remote -a fx-breach-alerts`
+
+Then, you can push your local unmerged branch to Heroku:
+
+* `git push -f heroku change-1:main`
+
+Merges to main will fail to deploy until someone manually resets it to `main`:
+
+* `git push -f heroku main`
+
+
+## Release to Stage
+Every tag pushed to GitHub is automatically deployed to the [Stage][stage]
+server. The standard practice is to create a tag from `main` every Tuesday at
+the end of the day, and to name the tag with `YYYY-MM-DD` [CalVer][calver]
+syntax. This tag will include only the changes that have been merged to `main`.
+E.g.,
+
+1. `git tag 2022.08.02`
+2. `git push --tags`
```

Comment:
I'm not sure that it's better, but just offering up for consideration: I tend to push a specific tag, just in case I have inadvertently created other tags locally. On the other hand, pushing all tags every time might be a better way of finding out that you've done that.

```suggestion
2. `git push origin 2022.08.02`
```